espressif/idf-eclipse-plugin
# IEP-813 Debugger: Update the hardware-watchpoint-limit based on the selection of the target

## Description

The default command for setting hardware-watchpoint-limit was changed in a way, that allows us to adjust it based on the target selection. For example, esp32 has 2 breakpoints/watchpoints and esp32-c3 has 8 hardware breakpoints/watchpoints

Fixes # ([IEP-813](https://jira.espressif.com:8443/browse/IEP-813))

## Type of change

Please delete options that are not relevant.
- New feature (non-breaking change which adds functionality)


## How has this been tested?

Test 1:
- Create a new debug configuration
- edit the new configuration, go to the Startup tab. The new initialization commands line should look like this: 
<img width="830" alt="Screenshot 2023-03-28 at 19 34 35" src="https://user-images.githubusercontent.com/24419842/228308435-f6317cfd-c665-4ea9-b455-172e0f3807ab.png">

Test 2:
- Debug the project with different boards. It should be a different number of hardware watchpoints based on the built idf target. The numbers are next: esp32="2", esp32s2="2", esp32s3="2", esp32c3="8", esp32c2="2", "esp32c6"="4", "esp32h2"="8".
- You can check the command in the log after debugging. It will be a message like this: `!MESSAGE Initialization Commands:: mon reset halt
flushregs
set remote hardware-watchpoint-limit 2`


**Test Configuration**:
* ESP-IDF Version:
* OS (Windows,Linux and macOS):

## Dependent components impacted by this PR:

- Startup Tab (Initialization Commands)
- Debug

## Checklist
